components/style/attr.rs

```diff
@@ -32,13 +33,11 @@ pub enum AttrValue {
     Url(DOMString, Option<Url>),
 }
 
-/// Shared implementation to parse an integer according to
-/// <https://html.spec.whatwg.org/multipage/#rules-for-parsing-integers> or
-/// <https://html.spec.whatwg.org/multipage/#rules-for-parsing-non-negative-integers>
-fn do_parse_integer<T: Iterator<Item=char>>(input: T) -> Result<i64, ()> {
-    let mut input = input.skip_while(|c| {
-        HTML_SPACE_CHARACTERS.iter().any(|s| s == c)
-    }).peekable();
+fn do_parse_integer<T>(input: T) -> Result<i64, ()>
```

Comment:
Perhaps this should take in a `&DOMString`, just to be consistent with `parse_double`.